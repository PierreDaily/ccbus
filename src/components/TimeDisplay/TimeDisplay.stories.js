import DisplayTime from './index.vue';

// More on default export: https://storybook.js.org/docs/vue/writing-stories/introduction#default-export
export default {
  title: 'DisplayTime',
  component: DisplayTime,
  // More on argTypes: https://storybook.js.org/docs/vue/api/argtypes
  argTypes: {
    time: {control: { type: 'select' },
    options: ['9:30 am', '12:00 pm'],
  },}
};

// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args) => ({
  // Components used in your story `template` are defined in the `components` object
  components: { DisplayTime },
  // The story's `args` need to be mapped into the template through the `setup()` method
  setup() {
    return { args };
  },
  // And then the `args` are bound to your component with `v-bind="args"`
  template: '<display-time v-bind="args" />',
});

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
Primary.args = {
    time: '11:30 am',
};
