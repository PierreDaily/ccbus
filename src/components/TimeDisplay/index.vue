<template>
  <div>
    <img :src="clockIcon" />
    <span>{{time || '--:--'}}</span>
  </div>
</template>

<script lang="ts">

import clockIcon from "@/assets/icons/clock.svg";
export default {
  name: "TimeDisplay",
  props: {
    time: String
  },
  setup() {
    return {
      clockIcon,
    };
  },
};
</script>

<style lang="scss" scoped>
div {
  height: 20px;
  width: 90px;
}

span {
  color: var(--font-color-black);
  display: inline-block;
  font-family: var(--font-primary);
  font-size: 18px;
  font-style: normal;
  font-weight: 400;
  line-height: 21px;
  text-align: right;
}

img {
  height: 10px;
  left: -2px;
  position: relative;
  top: -4px;
  width: 10px;
}
</style>
