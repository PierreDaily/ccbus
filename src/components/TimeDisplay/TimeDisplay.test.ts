import { mount, shallowMount } from "@vue/test-utils";
import TimeDisplay from "./index.vue";

const time = "12:00";

test("Mount component", async () => {
  // arrange
  expect(TimeDisplay).toBeTruthy();


  const wrapper = mount(TimeDisplay, {
    props: {
        time
    },
  });

  // assert

  expect(wrapper.html()).toMatchSnapshot();
});

test("Should render 12:00 when time props is passed on", async () => {
  // arrange
  expect(TimeDisplay).toBeTruthy();

  const wrapper = mount(TimeDisplay, {
    props: {
        time
    },
  });

  // assert
  expect(wrapper.text()).toContain(time);

});


test("Should render '--:--' when time props isn't passed on", async () => {
    // arrange
    expect(TimeDisplay).toBeTruthy();
  
    const wrapper = mount(TimeDisplay, {
      props: {
      },
    });
  
    // assert
    expect(wrapper.text()).toContain("--:--");
  
  });
