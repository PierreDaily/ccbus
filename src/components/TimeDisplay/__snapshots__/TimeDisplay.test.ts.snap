// Vitest Snapshot v1

exports[`Mount component 1`] = `"<div data-v-015bd910=\\"\\"><img src=\\"/@fs/src/assets/icons/clock.svg\\" data-v-015bd910=\\"\\" /><span data-v-015bd910=\\"\\">12:00</span></div>"`;
